src/commands.rs

```diff
@@ -35,6 +43,42 @@ pub struct CommonOptions {
 }
 
 impl CommonOptions {
+    /// Get the signing key for the specified user and registry.
+    pub fn get_signing_key(&self) -> Result<signing::PrivateKey> {
```

Comment:
This probably needs a closer look; i'm not sure i am using the `default_url` from the `Config` correctly here.

Reply:
I think we can probably remove this method (see my other comment).